src/sstruct_mv/sstruct_matrix.c

```diff
@@ -823,10 +823,12 @@ hypre_SStructPMatrixGetDiagonal( hypre_SStructPMatrix  *pmatrix,
  *==========================================================================*/
 
 /*--------------------------------------------------------------------------
+ * hypre_SStructUMatrixInitialize
  *--------------------------------------------------------------------------*/
 
 HYPRE_Int
-hypre_SStructUMatrixInitialize( hypre_SStructMatrix *matrix )
+hypre_SStructUMatrixInitialize( hypre_SStructMatrix  *matrix,
```

Comment:
Just wondering if memory_location should be set through a 'Set' function instead.  We should parallel what is done in parcsr, and maybe that's what you've already done here.

Reply:
In parcsr, we have the "_v2" functions that take the memory location as input argument. I think having a separate SetMemoryLocation is a good idea